ftw/zipexport/generation.py

```diff
@@ -50,6 +58,14 @@ def generate_unique_filepath(self, file_path):
             if new_filename not in self.zip_file.namelist():
                 return new_filename
 
+    def check_disk_has_space_for_file(self, file_d):
+        disk_stat = os.statvfs(self.tmp_file.name)
+        bytes_free = disk_stat.f_frsize * disk_stat.f_bavail
+        file_d.seek(0, os.SEEK_END)
+        file_size = file_d.tell()
+        file_d.seek(0)
```

Comment:
If the ``file_d`` pointer was not at ``0`` initially, this would set it to ``0``.
Although in most situations this should not be a problem, I would preserve the pointer position, e.g. like this:
```python
position = file_d.tell()
file_d.seek(0, os.SEEK_END)
file_size = file_d.tell() - position
file_d.seek(position)
```